receiver/http.go

```diff
@@ -57,12 +61,30 @@ type HTTP struct {
 type receiver struct {
 	Handler  *skogul.Handler
 	settings *HTTP
+	auth     *HTTPAuth
 }
 
 type httpReturn struct {
 	Message string
 }
 
+func (auth *HTTPAuth) auth(r *http.Request) (bool, error) {
+	var authErrMsg skogul.Error
+
+	// If username is present we assume username/pw auth
+	if auth.Username != "" {
```

Comment:
Go best-practice would have you invert this to "if auth.Username == "" { return false, nil }". Not a huge deal.

On a different note: as long as auth only has username/password, we should ensure that either none or both are specified. In the existing implementation this is done on startup, but should be done in Verify() in the future.

Reply:
My thought process regarding the first point here was to be able to re-use part of this logic for the upcoming client cert validation feature, so I didn't want to return early until all auth checks were exhausted.
Will fix second point.